### PR TITLE
loosen duration parsing

### PR DESCRIPTION
**Pull request by mkadin**
we're getting errors and losing money for durations that are 00:00:60

I'm sure I fucked something up so look at it

### CODE REVIEW

File: duration_test.go [hunk 1]
```diff
@@ -63,9 +63,35 @@ func TestDurationUnmarshal(t *testing.T) {
...
+	if assert.NoError(t, d.UnmarshalText([]byte("1:10.100"))) {
+		assert.Equal(t, Duration(1*time.Minute+10*time.Second+100*time.Millisecond), d)
+	}
+	if assert.NoError(t, d.UnmarshalText([]byte("00:1:10.100"))) {
+		assert.Equal(t, Duration(1*time.Minute+10*time.Second+100*time.Millisecond), d)
+	}
+
+	assert.EqualError(t, d.UnmarshalText([]byte("00:00:00.-1")), "invalid duration -- invalid milliseconds: 00:00:00.-1")
+	assert.EqualError(t, d.UnmarshalText([]byte("00:00:00.1000")), "invalid duration -- milliseconds too long: 00:00:00.1000")
+	assert.EqualError(t, d.UnmarshalText([]byte("00h01m")), "invalid duration -- invalid time value: 00h01m 00h01m")
```
Comment (BrianMargolis) [inline duration_test.go:108]: you can also test the empty milliseconds case ("00:00:12.") and the too-many-periods can ("00:00:12.34.56")

File: duration_test.go [hunk 2]
```diff
@@ -63,9 +63,35 @@ func TestDurationUnmarshal(t *testing.T) {
...
+	}
+	if assert.NoError(t, d.UnmarshalText([]byte("1:10.1"))) {
+		assert.Equal(t, Duration(1*time.Minute+10*time.Second+100*time.Millisecond), d)
+	}
+	if assert.NoError(t, d.UnmarshalText([]byte("1:10.100"))) {
+		assert.Equal(t, Duration(1*time.Minute+10*time.Second+100*time.Millisecond), d)
+	}
+	if assert.NoError(t, d.UnmarshalText([]byte("00:1:10.100"))) {
+		assert.Equal(t, Duration(1*time.Minute+10*time.Second+100*time.Millisecond), d)
+	}
```
Comment (BrianMargolis) [inline duration_test.go:92]: could also test mm:ss and mm:ss.millis

**Review by BrianMargolis — APPROVED** [review]
read it carefully and the logic looks right, just a few suggestions on test cases to add

**Comment by mkadin** [comment]
proudly written by a human
